apps/05Mastermind.js

```diff
@@ -20,20 +22,60 @@ function generateSolution() {
         var randomIndex = getRandomInt(0, letters.length);
         solution += letters[randomIndex];
     }
+
 }
+var solution = generateSolution();
 
 function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
-    // your code here
+function generateHint(solution, guess) {
+
+  var correctLetterLocations =  0;
+  var guessArray = guess.split('');
+  var solutionArray = solution.split('');
+
+  for (var i = 0; i < solutionArray.length; i++) {
+    if (solutionArray[i] === guessArray[i]) {
+        correctLetterLocations++;
+        solutionArray[i] = null;
+}
 }
+  var correctLetters = 0;
 
-function mastermind(guess) {
-    // your code here
+  for (i = 0; i < solutionArray.length; i++) {
+    var targetindex = (guessArray[i] === solutionArray.indexOf())
```

Comment:
Hmm this indexOf() call doesn't look complete to me. Remember when we call `arr.indexOf(el)` we're looking for `el` somewhere inside `arr`